src/Decorator/AbstractDecorator.php

```diff
@@ -0,0 +1,40 @@
+<?php
+
+namespace Phlib\Logger\Decorator;
+
+use Psr\Log\AbstractLogger;
+use Psr\Log\LoggerInterface;
+
+/**
+ * @package Phlib\Logger
+ */
+abstract class AbstractDecorator extends AbstractLogger
+{
+    /**
+     * @var LoggerInterface
+     */
+    private $logger;
+
+    /**
+     * @var mixed
+     */
+    protected $config;
```

Comment:
I was about to write

> Downside then is that every concrete has to implement a construct. Something like the DefaultContext decorator currently wouldn't have to.

But then realised that actually, even the DefaultContext would want a constructor to ensure that the config parameter is an array and not anything else.

OK, so agreed in the Abstract but not used? With explanation that concrete should implement if required.